File: modules/agent/src/schemas/partnership.py
"""Partnership schemas for API validation."""

from typing import List, Optional

from pydantic import BaseModel


class PartnershipCreate(BaseModel):
    account_type: str = "Organization"
    account: Optional[str] = None
    contacts: Optional[List[dict]] = None
    industry: Optional[List[str]] = None
    industry_ids: Optional[List[int]] = None
    partnership_name: str
    partnership_type: Optional[str] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    description: Optional[str] = None
    status: str = "Exploring"
    source: str = "manual"
    project_ids: Optional[List[int]] = None


class PartnershipUpdate(BaseModel):
    account: Optional[str] = None
    contacts: Optional[List[dict]] = None
    partnership_name: Optional[str] = None
    partnership_type: Optional[str] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    source: Optional[str] = None
    project_ids: Optional[List[int]] = None

File: modules/agent-go/internal/schemas/partnership.go
package schemas

// PartnershipCreate represents the request body for creating a partnership
type PartnershipCreate struct {
	AccountType     string    `json:"account_type" validate:"omitempty,oneof=Organization Individual"`
	Account         *string   `json:"account"`
	Contacts        []Contact `json:"contacts"`
	Industry        []string  `json:"industry"`
	IndustryIDs     []int64   `json:"industry_ids"`
	Title           string    `json:"title" validate:"required"`
	PartnershipName string    `json:"partnership_name" validate:"required"`
	PartnershipType *string   `json:"partnership_type"`
	StartDate       *string   `json:"start_date"`
	EndDate         *string   `json:"end_date"`
	Description     *string   `json:"description"`
	Status          string    `json:"status"`
	Source          string    `json:"source" validate:"omitempty,oneof=manual inbound referral event campaign agent"`
	ProjectIDs      []int64   `json:"project_ids"`
}

// PartnershipUpdate represents the request body for updating a partnership
type PartnershipUpdate struct {
	Account         *string   `json:"account"`
	Contacts        []Contact `json:"contacts"`
	Title           *string   `json:"title"`
	PartnershipName *string   `json:"partnership_name"`
	PartnershipType *string   `json:"partnership_type"`
	StartDate       *string   `json:"start_date"`
	EndDate         *string   `json:"end_date"`
	Description     *string   `json:"description"`
	Status          *string   `json:"status"`
	Source          *string   `json:"source"`
	ProjectIDs      []int64   `json:"project_ids"`
}
